mate-panel/panel-addto.c

```diff
@@ -565,62 +572,57 @@ panel_addto_prepend_alias (GSList         **parent_list,
 			   MateMenuTreeAlias  *alias,
 			   const char      *filename)
 {
-	MateMenuTreeItem *aliased_item;
-
-	aliased_item = matemenu_tree_alias_get_item (alias);
-
-	switch (matemenu_tree_item_get_type (aliased_item)) {
+	switch (matemenu_tree_alias_get_aliased_item_type (alias)) {
 	case MATEMENU_TREE_ITEM_DIRECTORY:
 		panel_addto_prepend_directory (parent_list,
-					       MATEMENU_TREE_DIRECTORY (aliased_item),
-					       filename);
+				matemenu_tree_alias_get_directory(alias),
```

Comment:
Here and a few lines below - same `matemenu_tree_alias_*` functions as I mentioned before...